ScreenCapture/ScreenCapture.cpp

```diff
@@ -334,6 +345,115 @@ namespace WPEFramework
+            LOGWARN("VNCServerLogMessage called");
+        }
+
+        bool ScreenCapture::getScreenshotRealtek(std::vector<unsigned char> &png_out_data)
+        {
+            ErrCode err;
+            vnc_bool_t result;
+            vnc_uint8_t* buffer; 
+            VncServerFramebufferAPI api;
+
+            api.framebufferUpdateReady = FakeVNCServerFramebufferUpdateReady;
+            api.framebufferDetailsChanged = FakeVNCServerFramebufferDetailsChanged;
+            api.paletteChanged = FakeVNCServerPaletteChanged;
+            api.logMsg = FakeVNCServerLogMessage;
+
+            FBContext *context = NULL;
+            err = fbCreate(&context);
+            if (err != ErrNone) {
+                LOGERR("fbCreate fail");
+                return false;
+            }
+
+            err = fbInit(context, &api, NULL);
+            if (err != ErrNone) {
+                LOGERR("fbInit fail");
+                return false;
+            }
+
+            PixelFormat *pf = fbGetPixelFormat(context);
+            if(pf) {
+                LOGINFO("fbGetPixelFormat:");
+                LOGINFO("\tbitsPerPixel=%d", pf->bitsPerPixel );
+                LOGINFO("\tdepth=%d", pf->depth);
+                LOGINFO("\tbigEndian=%s", (pf->bigEndian)?"true":"false");
+                LOGINFO("\ttrueColour=%s", (pf->trueColour)?"true":"false");
+                LOGINFO("\tredMax=%d", pf->redMax);
+                LOGINFO("\tgreenMax=%d", pf->greenMax);
+                LOGINFO("\tblueMax=%d", pf->blueMax);
+                LOGINFO("\tredShift=%d", pf->redShift);
+                LOGINFO("\tgreenShift=%d", pf->greenShift);
+                LOGINFO("\tblueShift=%d", pf->blueShift);
+            }
+
+            if (32 != pf->bitsPerPixel)
+            {
+                LOGERR("Unsupported bits per pixel: %d", pf->bitsPerPixel);
+                return false;
+            }
+
+            vnc_uint16_t w = fbGetWidth(context);
+            LOGINFO("fbGetWidth=%d", w);
+            vnc_uint16_t h = fbGetHeight(context);
+            LOGINFO("fbGetHeight=%d", h);
+            vnc_uint16_t s = fbGetStride(context);
+            LOGINFO("fbGetStride=%d", s);
+
+            buffer = fbGetFramebuffer(context);
+
+            if(buffer) {
+                LOGINFO("fbGetFramebuffer=ok"); 
+
+                for(unsigned int n = 0; n < h; n++)
+                {
+                    for(unsigned int i = 0; i < w; i++)
+                    {
+                        unsigned char *color = buffer + n * s + i * 4;
+
+                        unsigned char blue = color[0];
+                        color[0] =  color[2];
+                        color[2] = blue;
+                    }
+                }
+
+                if(!saveToPng(buffer, w, h, png_out_data))
+                {
+                    LOGERR("could not convert Nexus screenshot to png");
+                    return false;
+                }
+                else
+                    return true;
```

Comment:
I don't have documentation fo fbInit(), but since context is a simple pointer, and pointer to context is passed to fbInit() I assume it must be malloc()ing some memory. So each time you successfully savetopng() you will leak memory. You should add fbDestroy(context) here (and in all places of early return after fbInit()).